ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -7487,6 +7487,74 @@ void HqlCppTranslator::doBuildAssignExecuteWhen(BuildCtx & ctx, const CHqlBoundT
     }
 }
 
+void HqlCppTranslator::doBuildExprPow(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & tgt)
+{
+    assertex(expr->numChildren() == 2);
+    
+    IHqlExpression * right = expr->queryChild(1);
+
+    bool notIntegerPower = !right->queryType()->isInteger();
+    if (notIntegerPower)
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    IValue * value = right->queryValue();
+    bool notConstant = (value == nullptr);
+    if (notConstant) 
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    __int64 rightIntVal = value->getIntValue();
+    __int64 absExp = abs(rightIntVal);
+
+    const uint MAX_EXP_INLINE_VALUE = 3;
+    if (absExp > MAX_EXP_INLINE_VALUE)
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+    
+    OwnedITypeInfo realType = makeRealType(DEFAULT_REAL_SIZE);
+    CHqlBoundExpr leftBound;
+    
+    IHqlExpression * left = expr->queryChild(0);
+    OwnedHqlExpr castLeft = ensureExprType(left, realType);
+    buildTempExpr(ctx, castLeft, leftBound);
+
+    if (rightIntVal == 0)
+    {
+        tgt.expr.setown(createConstant(left->queryType()->castFrom(1.0)));
+    }
+    else if (rightIntVal == 1)
+    {
+        tgt.expr.setown(LINK(leftBound.expr));
+    }
+    else
+    {
+        LINK(leftBound.expr);
```

Comment:
This should be combined with the outExpr definition:

        IHqlExpression* outExpr = LINK(leftBound.expr);

But better is to use an OwnedHqlExpr so that if an exception is thrown it will be cleanly released.